scripts/migration/schema-0.2-GPII-4014/createSimulatedGpiiKeys.js

```diff
@@ -0,0 +1,211 @@
+    options.newDocs = [];
+    options.numOfCreatedKeys = 0;
+    options.parsedCouchDbUrl = url.parse(options.couchDbUrl);
+    options.postOptions = {
+        hostname: options.parsedCouchDbUrl.hostname,
+        port: options.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        auth: options.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // IMPORTANT: FILL IN PER REQUEST
+            "Content-Type": "application/json"
+        }
+    };
+    console.log("COUCHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+    return options;
+};
+
+/**
+ * Generate an array of new documents to create.
+ * @param {Number} numOfKeys - The number of new GPII keys to create
+ * @return {Array} - An array of GPII keys and corresponding prefs safe documents to create.
+ */
+gpii.migration.GPII4014.generateKeyData = function (numOfKeys) {
+    var newDocs = [];
+    for (var i = 0; i < numOfKeys; i++) {
+        var currentTime = new Date().toISOString();
+        var gpiiKeyId = uuid.v4();
+        var prefsSafeId = "prefsSafe-" + gpiiKeyId;
+        var prefs = {
+            "flat": {
+                "contexts": {
+                    "gpii-default": {
+                        "name": "Default preferences",
+                        "preferences": {
+                            "http://registry.gpii.net/common/language": "en-US"
+                        }
+                    }
+                }
+            }
+        };
+
+        var newGpiiKey = {
+            "_id": gpiiKeyId,
+            "type": "gpiiKey",
+            "schemaVersion": "0.1",
+            "prefsSafeId": prefsSafeId,
+            "prefsSetId": "gpii-default",
+            "revoked": false,
+            "revokedReason": null,
+            "timestampCreated": currentTime,
+            "timestampUpdated": null
+        };
+
+        var newPrefsSafe = {
+            "_id": prefsSafeId,
+            "type": "prefsSafe",
+            "schemaVersion": "0.1",
+            "prefsSafeType": "user",
+            "name": gpiiKeyId,
+            "password": null,
+            "email": null,
+            "preferences": prefs,
+            "timestampCreated": currentTime,
+            "timestampUpdated": null
+        };
+        newDocs.push(newGpiiKey);
+        newDocs.push(newPrefsSafe);
+    };
+    return newDocs;
+};
+
+/**
+ * Log how many GPII documents were updated.
+ * @param {String} responseString - Response from the database (ignored)
+ * @param {Object} options - Object containing the set of documents:
+ * @param {Array} options.numOfKeysToCreate - Total number of GPII keys to create.
+ * @param {Array} options.numOfCreatedKeys - Total number of GPII keys that have been created.
```

Comment:
Not an Array but a number. Also, explain clearly that this option will be written by a call to the function. It might be a bit more moral to have this the promise yield rather than a side-effect, and have the message created by the caller

Reply:
Bump coment